### PR TITLE
Add/fix ModelParallel layout map for qwen3_5_moe

### LINKED ISSUE #2838: Add layout map for Qwen3.5-MoE (hybrid linear/full attention)
Qwen3.5-MoE mixes full-attention and linear-attention (GatedDeltaNet) blocks in a MoE model, and it has no layout map. Adding one means the standard QKV/embedding/expert-bank column-row rules plus explicit handling for the linear-attention path: `in_proj_qkv`/`in_proj_z` column-parallel, `out_proj` row-parallel, while the tiny `in_proj_a`/`in_proj_b`, the depthwise conv kernel, and the 1-D `dt_bias`/`A_log` stay replicated. The shared-expert gate is left replicated too since its output dim is 1. QKV are `(hidden, num_heads, head_dim)` here, so hidden on data and heads on model; GQA k/v replicated.

One limitation: the vision encoder and interleave weights aren't sharded, only the text decoder. The mesh-shape and live-preset sweeps each force a two-layer hybrid stack so both token-mixer rule sets get asserted. Tracked in #2822, under #2807.

### PR DESCRIPTION
**Pull request by pctablet505**
Closes #2838

Part of https://github.com/keras-team/keras-hub/issues/2807. Depends on #2809 (`TestCase.run_distribution_test`) for CI to pass -- this PR's own diff is independent (based on master, not stacked), so it stays small and reviewable regardless of merge order.

## Summary

Adds `Qwen3_5MoeBackbone.get_layout_map` so `Qwen3_5MoeBackbone` (a hybrid full-attention/linear-attention MoE model) can be sharded across a device mesh with `keras.distribution.ModelParallel`, following the same Megatron-style column/row-parallel convention used elsewhere in this series (e.g. Gemma, Qwen3):

- **QKV / reverse-embedding sharding axes.** `query`/`key`/`value` kernels here are `(hidden_dim, num_heads, head_dim)` (contracting `hidden_dim` first), not Gemma's `(num_heads, hidden_dim, head_dim)`. `hidden_dim` shards on the data axis and heads shard on the model axis -- the correct orientation for Megatron-style tensor parallelism given this weight layout. `token_embedding/reverse_embeddings` (the `(hidden, vocab)` output projection) gets its own transposed spec rather than reusing the input embedding's tuple.
- **Linear-attention (GatedDeltaNet) projections.** `in_proj_qkv` and `in_proj_z` (column-parallel) and `out_proj` (row-parallel) have explicit layout rules. The tiny `in_proj_a`/`in_proj_b` (project to `num_v_heads`, a small axis), the depthwise `conv1d_kernel`, and the 1-D `dt_bias`/`A_log` are intentionally left replicated (documented in the code and covered by the test's `allow_replicated` list).
- **Shared-expert gate** (`(hidden_dim, 1)`) is intentionally left replicated -- its output dim is 1, so there's nothing to shard on the model axis.
- Key/value kernels stay replicated on the model axis (GQA `num_key_value_heads` is independent of and typically much smaller than `num_query_heads`; sharding it would raise an `IndivisibleError` whenever the model-mesh dimension doesn't divide it evenly), while `hidden_dim` still shards on the data axis.
- MoE expert bank (`expert_feedforward_gate_dense`/`expert_feedforward_output_dense`), router, and shared-expert dense layers use column/row-parallel sharding.

## Limitations

- The vision encoder and `interleave_embeddings` weights (used when a `vision_encoder` is attached for multimodal inputs) are not covered by any layout rule and stay fully replicated. `get_layout_map` only shards the text-decoder stack (attention/linear-attention/MoE/embeddings); sharding the vision tower is out of scope for this PR.
- The model-parallel mesh axis can't usefully exceed a preset's `num_query_heads` -- that's the axis actually being tensor-parallelized here -- see [the design doc's "hard limit" section](https://github.com/keras-team/keras-hub/issues/2807#issuecomment-5006606419) for the full reasoning. For this model's real presets:

  | preset | num_query_heads | safe model-axis up to | first failing model-axis |
  |---|---|---|---|
  | `qwen3_5_moe_35b_a3b` | 16 | 16 | 32 |
  | `qwen3_5_moe_35b_a3b_base` | 16 | 16 | 32 |

  Both presets use `num_key_value_heads=2`, so any mesh past model-axis=16 fails on the query/attention-output kernels specifically, not on `hidden_dim` or the MoE expert bank -- those stay divisible well past this range for both presets.

## Tests

`qwen3_5_moe_backbone_test.py` adds test tiers mirroring the pattern already reviewed for Gemma (`gemma_backbone_test.py`):

- **Tier 1** (`test_distribution`): exact end-to-end sharding-spec + full-weight-coverage + forward/backward/optimizer-step + numerical-parity/convergence check via `TestCase.run_distribution_test` from #2809, on a real 2-device mesh (pinned to exactly 2, not `len(devices)`, so the default config's `num_key_value_heads=2` is exercised deterministically). Uses `is_moe=True` for the looser MoE parity tolerance and asserts both the full-attention and linear-attention (GatedDeltaNet) sublayer specs, since the default 4-layer test config includes both layer types.
- **Tier 2** (`test_layout_map_mesh_shapes`): CI-safe sweep over `CAPPED_MESH_SHAPES` (`(2,4)`, `(1,8)`, `(4,4)`, `(2,2,2)`, `(1,1,8)`, `(2,2,4)` -- capped at 16 total simulated devices) crossed with two synthetic width classes (`QWEN3_5_MOE_35B_A3B_DIMS`, `QWEN3_5_MOE_SMALL_DIMS`) that preserve the real preset's query:kv head ratio (8:1 GQA) and linear key:value head ratio (1:2), scaled down ~20-30x from the real `Qwen3.5-35B-A3B`-class config. Each sweep case builds a two-layer hybrid stack -- one `linear_attention` layer and one `full_attention` layer -- so both token-mixer rule sets are asserted in every (width, mesh) combo, not just at the fixed 2-device mesh in Tier 1 (layout rules are per-decoder-block regexes, so depth beyond one-of-each doesn't affect spec matching), and asserts full-weight-coverage via a shared helper. Skips (not fails) any `(width, mesh)` combo where `num_query_heads` or the linear-attention fused `in_proj_qkv` width doesn't divide the mesh's model axis -- an inherent tensor-parallelism limit, not a bug.
- **Tier 3** (`test_layout_map_live_presets`, opt-in via `kaggle_key_required` + `multi_device` + `extra_large` markers): fetches every real preset's `config.json` (metadata only, no weights), dedupes by divisibility-relevant dims, forces the same two-layer hybrid stack as Tier 2 onto each width class, and for each unique width class estimates its bf16 two-decoder-block + expert-bank build footprint before attempting a build, skipping (via `self.subTest` + a logged reason) any combo that would exceed a 300MB local safety threshold.

### Verification

Real local test results (this branch's own base, `master`, doesn't yet include #2809's `run_distribution_test` helper, so these were run against a temporary checkout with #2809's helper commit applied on top of this PR's diff -- the diff itself is unchanged and unstacked):

```
============================= test session starts ==============================
platform linux -- Python 3.13.13, pytest-9.1.1, pluggy-1.6.0
plugins: anyio-4.14.2, cov-7.1.0, xdist-3.8.0, jaxtyping-0.3.11
created: 2/2 workers
2 workers [20 items]

keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_backbone_basics PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_auxiliary_loss PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_distribution PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_1x1x8 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_1x8 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_2x2x2 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_2x2x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_2x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_35b_a3b_mesh_4x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_1x1x8 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_1x8 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_2x2x2 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_2x2x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_2x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_mesh_shapes_qwen3_5_moe_small_mesh_4x4 PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_num_parameters PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_session PASSED
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_saved_model SKIPPED (needs --run_large)
keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py::Qwen3_5MoeBackboneTest::test_layout_map_live_presets SKIPPED (needs --run_extra_large / kaggle key)

================== 17 passed, 3 skipped in 333.92s (0:05:33) ===================
```

Run with `KERAS_BACKEND=jax XLA_FLAGS=--xla_force_host_platform_device_count=16 python -m pytest keras_hub/src/models/qwen3_5_moe/qwen3_5_moe_backbone_test.py -v -n 2`, capped at 16 simulated CPU devices per the mesh-shape matrix. Both skips are the expected default-config skips (`test_saved_model` needs `--run_large`; `test_layout_map_live_presets` needs `--run_extra_large` + Kaggle credentials and is intentionally not exercised in this pass by design -- see the module comment above `CAPPED_MESH_SHAPES`).

`ruff check` / `ruff format --check`: clean on both changed files.

### CODE REVIEW

**Comment by pctablet505** [comment]
@gemini-code-assist review

**Comment by pctablet505** [comment]
@gemini-code-assist review
